### ISSUE TITLE
Weights are not updated correctly using GPU

### ISSUE THREAD

**Issue by renruoxu** [fchollet/keras#634]
I am training MLP using GPU mode. The loss value keeps increasing and accuracy is extremely low (~9%). However, when I switch to CPU mode, everything is fine (loss decreases and accuracy increases ~95%). 

There shouldn't be problem with Theano, as training using theano MLP codes in GPU mode is working fine. I'm using CUDA 7.0 and Theano 0.7.0. 

Anyone has the same issue?

**Comment by rodrigob** [comment]
Same as #511 and #617

**Comment by fchollet** [comment]
Updating to latest Keras and Theano version fixes the issue (#511).
